src/components/editor/Bindings/SchemaInference/Dialog.tsx

```diff
@@ -0,0 +1,442 @@
+                            schemaUpdateErrored,
+                            setSchemaUpdateErrored,
+                            setCollectionData,
+                            setOpen
+                        );
+                    }
+                }
+
+                setLoading(false);
+            } else {
+                setSchemaUpdateErrored(true);
+            }
+        },
+    };
+
+    return currentCollection ? (
+        <Dialog
+            open={open}
+            maxWidth="lg"
+            aria-labelledby={TITLE_ID}
+            sx={{
+                '& .MuiPaper-root.MuiDialog-paper': {
+                    backgroundColor: glassBkgWithoutBlur[theme.palette.mode],
+                    borderRadius: 5,
+                },
+            }}
+        >
+            <DialogTitle
+                component="div"
+                sx={{ display: 'flex', alignItems: 'center' }}
+            >
+                <DataObject />
+
+                <Typography variant="h6" sx={{ ml: 1 }}>
+                    <FormattedMessage id="workflows.collectionSelector.schemaInference.header" />
+                </Typography>
+            </DialogTitle>
+
+            <DialogContent>
+                <Typography sx={{ mb: 2 }}>
+                    <FormattedMessage id="workflows.collectionSelector.schemaInference.message" />
+                </Typography>
+
+                <Typography sx={{ mb: 4 }}>
+                    <FormattedMessage
+                        id="workflows.collectionSelector.schemaInference.message.schemaDiff"
+                        values={{ documents_read: documentsRead ?? 0 }}
+                    />
+                </Typography>
+
+                {schemaUpdateErrored ? (
+                    <AlertBox
+                        severity="error"
+                        short
+                        title={
+                            <Typography>
+                                <FormattedMessage id="workflows.collectionSelector.schemaInference.alert.generalError.header" />
+                            </Typography>
+                        }
+                    >
+                        <MessageWithLink messageID="workflows.collectionSelector.schemaInference.alert.patchService.message" />
+                    </AlertBox>
+                ) : null}
+
+                {documentsRead && documentsRead < DOCUMENT_THRESHOLD ? (
+                    <AlertBox
+                        severity="warning"
+                        short
+                        title={
+                            <Typography>
+                                <FormattedMessage id="workflows.collectionSelector.schemaInference.alert.lowDocumentCount.header" />
+                            </Typography>
+                        }
+                    >
+                        <FormattedMessage id="workflows.collectionSelector.schemaInference.alert.lowDocumentCount.message" />
+                    </AlertBox>
+                ) : null}
```

Comment:
This is fine... but I think this is a spot where we might want to just have a single error/alert. Also... could be a good spot for a stand alone component.